### PR TITLE
[RFR] Admin - Load config

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: admin/src/fetch/fetchConfig.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { Request } from './fetch';
+import fetchRequest from './fetchRequest';
+
+export const getEnvironmentRequest = ({ projectId, environmentName, configName = 'default', token } = {}) => {
```
Comment (ThieryMichel) [inline admin/src/fetch/fetchConfig.js]: getFetchConfigRequest ?

File: admin/src/utils/wrapWithLoading.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export default cb => (effects, ...args) =>
+    effects
+        .setLoading(true)
+        .then(() => cb(effects, ...args))
+        .then(value => effects.setLoading(false).then(() => value));
```
Comment (ThieryMichel) [inline admin/src/utils/wrapWithLoading.js]: and if cb reject ?
Add a `.catch` that setLoading and then rethrow the error.

File: admin/src/utils/wrapWithErrorHandling.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default cb => (effects, ...args) => {
+    return cb(effects, ...args).then(value => value, error => effects.setError(error.message));
```
Comment (ThieryMichel) [inline admin/src/utils/wrapWithErrorHandling.js]: Add unit test, and what if cb does not return a promise, or throw an error directly ?

File: admin/src/utils/wrapWithLoading.js (hunk 2 above)
Comment (ThieryMichel) [inline admin/src/utils/wrapWithLoading.js]: And add unit test

File: admin/src/utils/wrapWithErrorHandling.js (hunk 3 above)
Comment (djhi) [inline admin/src/utils/wrapWithErrorHandling.js]: It has to return a promise

**Comment by djhi** [comment]
> Maybe we should put the error state and effects only on the root state ?

Let's keep that for another PR until we are sure of what we can do :)

File: admin/src/utils/wrapWithErrorHandling.js [hunk 4]
```diff
@@ -1,3 +1,12 @@
 export default cb => (effects, ...args) => {
-    return cb(effects, ...args).then(value => value, error => effects.setError(error.message));
+    try {
+        const result = cb(effects, ...args);
```
Comment (ThieryMichel) [inline admin/src/utils/wrapWithErrorHandling.js]: you can wrap cb(effects, ...args) in a `Promise.resolve` and avoid the test for `then`

File: admin/src/utils/wrapWithErrorHandling.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+export default cb => (effects, ...args) => {
+    try {
+        const result = cb(effects, ...args);
```
Comment (ThieryMichel) [inline admin/src/utils/wrapWithErrorHandling.js]: you can wrap cb(effects, ...args) in a Promise.resolve and avoid the test for then

**Pull request merged**